### DISCUSSION TITLE
Using Static Data at App (Global) Level

### DISCUSSION THREAD

**Discussion by saleebm** [vercel/next.js#20237] [Help] [answered]
Hello, thank you all for your contributions to this awesome framework.

I am wondering if there is a way to query data that is statically optimized from outside of the page. I know there used to be getInitialProps but from my understanding, that blocks static optimization. 

The goal for this would be to provide globally scoped components data and optimize it into the app. The app could also revalidate the response, but it would be opt-in from the config to revalidate globally queried data. 

My use case might help clarify the practicality of this prop. I have my navigation menu on my layout component wrapping all pages that depends on dynamic data. Instead of moving that layout component around every page, and doing the same query on every page, it could just be one query outside of the app that returns that to the app root itself.

Thanks for your time and discussion.

**Comment by leerob** [comment, marked as answer]
The use of the data fetching methods `getStaticProps`, `getServerSideProps`, etc is for **top-level `pages/` only currently.**

https://nextjs.org/docs/basic-features/data-fetching

**Reply by saleebm** [reply]
Thanks

**Reply by saleebm** [reply]
for anyone else wondering, found this answer [here](https://github.com/vercel/next.js/discussions/11183#discussioncomment-114597) that works as well.

looking forward to be able to also use server components for this use case
